src/app/view/components/_common/dot-page-selector/service/dot-page-selector.service.ts

```diff
@@ -29,59 +29,217 @@ export interface DotPageAsset {
+                map((items: DotPageSeletorItem[]) => {
+                    return {
+                        data: items,
+                        type: 'page',
+                        query: searchParam.replace(HOST_FULL_REGEX, '')
+                    };
+                })
             );
     }
+
+    private getPagesSearchQuery(searchParam: string, hostId?: string): { [key: string]: {} } {
+        const parsedQuery = this.parseHost(searchParam);
+
+        let query = `+basetype:5 +path:*${parsedQuery[0]}*`;
+
+        query +=
+            parsedQuery.length > 1
+                ? ` +conhostName:*${parsedQuery[1]}*`
+                : hostId
+                ? ` +conhost:${hostId}`
+                : '';
+
+        return this.getRequestBodyQuery(query);
+    }
+
+    private getRequestBodyQuery(query: string): { [key: string]: {} } {
+        return {
+            query: {
+                query_string: {
+                    query: query
+                }
+            }
+        };
+    }
+
+    private getSites(param: string): Observable<DotPageSelectorResults> {
+        const query = `+contenttype:Host +host.hostName:*${this.getSiteName(param)}*`;
+
+        return this.getEsResults(query).pipe(
+            pluck('contentlets'),
+            flatMap((sites: Site[]) => sites),
+            map((site: Site) => {
+                return {
+                    label: `//${site.hostname}/`,
+                    payload: site,
+                    isHost: true
+                };
+            }),
+            toArray(),
+            map((items: DotPageSeletorItem[]) => {
+                return {
+                    data: items,
+                    type: 'site',
+                    query: param
+                };
+            })
+        );
+    }
+
+    private getSiteName(site: string): string {
+        return site.replace(/\//g, '');
+    }
+
+    private isAbsoluteUrl(param: string): boolean {
+        return URL_ABS_REGEX.test(param);
+    }
+
+    private isReSearchingForHost(param: string): boolean {
+        return this.currentHost && !this.itStartWithFullHost(param);
+    }
+
+    private itStartWithFullHost(param: string): boolean {
+        return HOST_FULL_REGEX.test(param);
+    }
+
+    private isTwoStepSearch(param): boolean {
+        return !this.currentHost && this.isAbsoluteUrl(param);
+    }
+
+    private parseHost(query: string): string[] {
+        const host = query.match(/^\/\/[^/]*\//g);
+        const search: string[] = [];
+        if (host) {
+            search.push(query.replace(host[0], '').replace(/\//g, '\\/'));
+            search.push(host[0].substr(2).slice(0, -1));
+            return search;
+        }
+        return [query.replace(/\//g, '\\/')];
+    }
+
+    private shouldSearchPages(param: string): boolean {
+        if (this.isReSearchingForHost(param)) {
+            this.currentHost = null;
+        }
+
+        return !!(this.currentHost || !param.startsWith('//'));
+    }
```

Comment:
maybe this service have a lot of private util method, I don't know if a better idea create a util class and move some of the private method to there